### ISSUE TITLE
Composer using GPT 3.5 under the hood, even after you select Claude 3.5 Sonnet

### ISSUE THREAD

**Issue by odragora** [getcursor/cursor#1844]
![image](https://github.com/user-attachments/assets/e67dae2c-8996-4cc6-a9fb-211050a080f3)

Details:

https://www.reddit.com/r/cursor/comments/1fabz3q/cursor_composer_using_gpt_35_under_the_hood_even/

**Comment by adgower** [comment]
@odragora can update the reddit post with link to this GH issue?

**Comment by arvid220u** [comment]
*TL;DR We're using the model you have selected + our custom-trained apply model; the reporting mentioning `gpt-3.5-turbo` is a remnant of old code.*

Hi! I understand that this is confusing.

Here's what's happening: Composer works by first asking the model you have selected (in this case, Claude 3.5 Sonnet) to write a plan for what files to edit (this is what you see streaming in). Then, we use our [custom-trained apply model](https://www.cursor.com/blog/instant-apply) to apply Sonnet's edit plan to the actual files.

For historical reasons, the apply request is called `slashEdit` in our code (it initially started out as a `/edit` command in the chat, and then evolved into the apply functionality). When it was a `/edit` command, it made sense to allow people to change the model that it was using, which is why we are sending up the `modelIntent` with the request. When the `slashEdit` request was co-opted for applying code blocks, it did not make sense to keep the model toggle (since often the model you want applying is not the same model you want giving you the plan). Hence, we started sending up just `gpt-3.5-turbo` as a default from the client, and then on our server we would override the model with our fast custom-trained apply model.

Hope this explains it! (Note that `gpt-3.5-turbo` is a very expensive model for its intelligence class these days, so it would never really make sense for us to use it...)

(Another note is that we just recently added a `composer` request event — previously, `composer` requests were marked as `chat`. We had a bug where our website did not know about the `composer` request event, and was instead just displaying empty brackets `{}`, which is what you see in the second request. We have now updated our website and you should see `composer` events in the list. Thanks for helping us catch that bug!)

**Comment by arvid220u** [comment]
We have now updated our event logging code to log the event as `fastApply` instead, to avoid confusion.

<img width="899" alt="Screenshot 2024-09-06 at 12 42 31 PM" src="https://github.com/user-attachments/assets/70d323d3-870f-42f7-b6f5-78c89d67f84d">
